connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala

```diff
@@ -230,55 +230,75 @@ private[client] object GrpcExceptionConverter {
         params.cause)),
     errorConstructor(params =>
       new NoSuchTableException(params.errorClass.orNull, params.messageParameters, params.cause)),
-    errorConstructor[NumberFormatException](params =>
+    errorConstructor(params =>
       new SparkNumberFormatException(
-        params.message,
-        params.errorClass,
+        params.errorClass.orNull,
         params.messageParameters,
         params.queryContext)),
+    errorConstructor[NumberFormatException](params =>
+      new SparkNumberFormatException(
+        errorClass = "_LEGACY_ERROR_TEMP_3104",
```

Comment:
Actually, this may need some special handling since params.errorClass and params.messageParameters may be set

Reply:
Maybe
```scala
errorClass = params.errorClass.orElse("_LEGACY_ERROR_TEMP_3104"),
messageParameters = params.errorClass match {
  case Some(_) => params.messageParameters
  case None => Map("message" -> params.message)
}
```

Reply:
How can it be set for **non**-`SparkThrowable` like here `NumberFormatException`?

Reply:
got it. will do.